src/lib/apis/stats/index.ts

```diff
@@ -0,0 +1,35 @@
+import { WEBUI_API_BASE_URL } from '$lib/constants';
+
+export const getStatsData = async (token?: string) => {
+	let error = null;
+
+	const headers: Record<string, string> = {
+		'Content-Type': 'application/json'
+	};
+
+	// Only add authorization header if token is provided
+	if (token) {
+		headers.Authorization = `Bearer ${token}`;
+	}
+
+	const res = await fetch(`${WEBUI_API_BASE_URL}/stats/`, {
```

Comment:
This mix of await / async syntax and use of the Promise api with a then and catch handler is not super idiomatic although I find it very readable.

Reply:
Here's a suggestion that's more consistent.

```typescript
let res;
try {
    const response = await fetch(`${WEBUI_API_BASE_URL}/stats/`, {
        method: 'GET',
        headers,
        credentials: 'include'
    });
    const data = await response.json();
    if (!response.ok) {
        throw data;
    }
    res = data;
} catch (err: any) {
    console.log(err);
    error = err.detail;
    res = null;
}
```